### PR TITLE
36 authorization unauthorized users cannot access frontend and send requests to backend

### PR DESCRIPTION
**Pull request by Jean28518**
- authorization should work now
- tests not updated yet
- our old authorization header became deprecated, spring boot is handling this for us.
- works with docker and with the old traditional setup

### CODE REVIEW

**Review by annikakrause — CHANGES_REQUESTED** [review]
The login doesn't work with every password, there might be problems with some special characters. It works when the password is only composed of characters and numbers. 
Example: "(2(%SfY;)C2§1@3o78,S" doesn't work, "2SfYC213o7S" does.

Edit: we tested it and the problem seems to be the "§" sign. If you can't find the reason why, this should be adjusted in the initial password generation and the change password page.

Apart from this, good work!

**Comment by valentinBraeutigam** [comment]
> The login doesn't work with every password, there might be problems with some special characters. It works when the password is only composed of characters and numbers. Example: "(2(%SfY;)C2§1@3o78,S" doesn't work, "2SfYC213o7S" does.

Some how the '§' breaks it. You're not able to login when you're password includes it. But I don't know why.

File: src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/security/DIUserPrincipal.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        Collection collection = new ArrayList<SimpleGrantedAuthority>();
+        collection.add(new SimpleGrantedAuthority(user.getUserRole().getAuthority()));
+        return collection;
+    }
+
+    @Override
+    public String getPassword() {
+        return "{noop}" + user.getPassword();
```
Comment (valentinBraeutigam) [inline src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/security/DIUserPrincipal.java:28]: What is this for? :) Just for my interest

**Review by valentinBraeutigam — CHANGES_REQUESTED** [review]
Looks good and works fine, good work :) '§' character has to be forbidden for passwords.

**Comment by valentinBraeutigam** [comment]
Tests have to be adjusted in the future since some of them are deprecated now and some don't run anymore since methods have been changed

File: src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/security/DIUserPrincipal.java (hunk 1 above)
Comment (Jean28518) [reply src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/security/DIUserPrincipal.java:28]: For that one :) https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/password/NoOpPasswordEncoder.html

**Comment by Jean28518** [comment]
Because of e.g. the '=' also the change password service won't work...

I now changed it to only allow !@#$%^&*  as special characters that we get no problems with any protocol in the future

**Comment by Jean28518** [comment]
Thank you very much for fixing the small problems and the merge conflict!

**Review by annikakrause — APPROVED** [review]
Good job :)

**Review by valentinBraeutigam — APPROVED** [review]
really good work :)

**Pull request merged**